### PR TITLE
Merge Develop into Master

### PR DESCRIPTION
**Pull request by Aplicticus**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1 @@
-# ShipServiceManagementService
```
Comment (markieo1) [inline README.md]: No need for this change

File: ShipServiceManagement.Logic/Interfaces/IShipServiceManager.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		/// <param name="shipService">The ship service.</param>
+		/// <returns></returns>
+		Task<ShipService> UpdateShipService(ShipService shipService);
+
+		/// <summary>
+		/// Deletes the ship service.
+		/// </summary>
+		/// <param name="name">The name.</param>
+		/// <returns></returns>
+		Task DeleteShipService(string name);
```
Comment (markieo1) [inline ShipServiceManagement.Logic/Interfaces/IShipServiceManager.cs]: Shouldn't this be based on the guid/id?

File: ShipServiceManagement.Logic/ShipServiceManagement.Logic.csproj [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (markieo1) [inline ShipServiceManagement.Logic/ShipServiceManagement.Logic.csproj]: Why is this project a netcore app. Should also work with .net standard ;-)

File: ShipServiceManagement.Messaging/ShipServiceManagement.Messaging.csproj [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (markieo1) [inline ShipServiceManagement.Messaging/ShipServiceManagement.Messaging.csproj]: Why is this project a netcore app. Should also work with .net standard ;-)

File: ShipServiceManagement.Models/ShipServiceManagement.Models.csproj [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (markieo1) [inline ShipServiceManagement.Models/ShipServiceManagement.Models.csproj]: Why is this project a netcore app. Should also work with .net standard ;-)

File: ShipServiceManagement.Persistence/Interfaces/IShipServiceService.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		/// <param name="price">The price.</param>
+		/// <returns></returns>
+		Task<ShipService> UpdateAsync(ShipService shipService);
+
+		/// <summary>
+		/// Deletes the asynchronous.
+		/// </summary>
+		/// <param name="name">The name.</param>
+		/// <returns></returns>
+		Task DeleteAsync(string name);
```
Comment (markieo1) [inline ShipServiceManagement.Persistence/Interfaces/IShipServiceService.cs]: To be based on guid

File: ShipServiceManagement.Persistence/Interfaces/IShipServiceService.cs [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+namespace ShipServiceManagement.Persistence.Interfaces
+{
+	public interface IShipServiceService
+	{
+		/// <summary>
+		/// Gets the asynchronous.
+		/// </summary>
+		/// <param name="name">The name.</param>
+		/// <returns></returns>
+		Task<ShipService> GetAsync(string name);
```
Comment (markieo1) [inline ShipServiceManagement.Persistence/Interfaces/IShipServiceService.cs]: To be based on guid

File: ShipServiceManagement.Persistence/Migrations/20180512125541_Initial.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    public partial class Initial : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "ShipService",
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    Name = table.Column<string>(nullable: true),
```
Comment (markieo1) [inline ShipServiceManagement.Persistence/Migrations/20180512125541_Initial.cs]: Name should also be `nullable: false`

File: ShipServiceManagement.Persistence/Properties/launchSettings.json [hunk 9]
```diff
@@ -0,0 +1,10 @@
+{
+  "profiles": {
```
Comment (markieo1) [inline ShipServiceManagement.Persistence/Properties/launchSettings.json]: This file should not exist

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,261 @@
+## Ignore Visual Studio temporary files, build results, and
```
Comment (markieo1) [inline .gitignore:1]: Correct the gitignore to this one: [VisualStudio.gitignore]( https://github.com/github/gitignore/blob/master/VisualStudio.gitignore)
That will solve some issues I have comments on now.

File: ShipServiceManagement.Persistence/ShipServiceManagement.Persistence.csproj [hunk 11]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (markieo1) [inline ShipServiceManagement.Persistence/ShipServiceManagement.Persistence.csproj]: Why is this project a netcore app. Should also work with .net standard ;-)

File: ShipServiceManagementService.sln [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ShipServiceManagement.Persistence", "ShipServiceManagement.Persistence\ShipServiceManagement.Persistence.csproj", "{1A4F48A2-FB36-430B-9C9C-85599688700D}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ShipServiceManagement.Models", "ShipServiceManagement.Models\ShipServiceManagement.Models.csproj", "{202656D3-7431-45A2-B2C3-20E32A4B1496}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ShipServiceManagement.Messaging", "ShipServiceManagement.Messaging\ShipServiceManagement.Messaging.csproj", "{6DDA6A38-BCC5-43C3-8BC8-456C8F58C734}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ShipServiceManagement.App", "ShipServiceManagement.App\ShipServiceManagement.App.csproj", "{9DDF49D8-EDBE-491C-AAF5-96C104FEEB3B}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{C0D8D9A7-D064-48DF-80B4-A5EFCFDC0280}"
```
Comment (markieo1) [inline ShipServiceManagementService.sln]: Do not add these files as solution items, since they might be non existant

**Review by markieo1 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Aplicticus) [reply README.md]: re-edited

Comment (markieo1) [reply README.md]: Still changed, shouldn't be

File: ShipServiceManagement.Models/ShipServiceManagement.Models.csproj [hunk 13]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Reference Include="System.ComponentModel.Annotations">
```
Comment (markieo1) [inline ShipServiceManagement.Models/ShipServiceManagement.Models.csproj]: Incorrect reference, should be package reference / nuget?

**Review by markieo1 — CHANGES_REQUESTED** [review]

**Review by markieo1 — APPROVED** [review]

**Pull request merged**